src/drt/src/TritonRoute.cpp

```diff
@@ -1040,6 +1040,14 @@ void TritonRoute::stackVias(odb::dbBTerm* bterm,
                             int bterm_bottom_layer_idx,
                             bool has_routing)
 {
+  // if pin access runs before the detailed route step, the nets with pins above
+  // the max routing layer will have the stacked vias inserted. added this check
+  // to avoid trying to stack new vias on the same nets.
+  odb::dbNet* net = bterm->getNet();
+  if (nets_with_stacked_vias_.count(net) == 1) {
+    return;
+  }
```

Comment:
What if PA was run in a separate process (like grt) and we are just opening the db?

Reply:
I've updated the code to look at the odb objects instead of relying on non-persistent information.